synapse/storage/databases/main/event_federation.py

```diff
@@ -749,7 +757,15 @@ def get_backfill_points_in_room_txn(
             # persisted in our database yet (meaning we don't know their depth
             # specifically). So we need to look for the approximate depth from
             # the events connected to the current backwards extremeties.
-            sql = """
+
+            if isinstance(self.database_engine, PostgresEngine):
+                least_function = "least"
+            elif isinstance(self.database_engine, Sqlite3Engine):
+                least_function = "min"
+            else:
+                raise RuntimeError("Unknown database engine")
```

Comment:
Maybe it is just me, but I really prefer these to be capitalized. I don't think we ever check both postgres and sqlite and then error, but shouldn't be a big deal.

```suggestion
            if isinstance(self.database_engine, PostgresEngine):
                least_function = "LEAST"
            elif isinstance(self.database_engine, Sqlite3Engine):
                least_function = "MIN"
            else:
                raise RuntimeError("Unknown database engine")
```

Reply:
Again, no strong opinions and I was deferring to the original implementation. (Though I prefer the `SHOUTING_CASE` for standard bits of SQL grammar like this)

Reply:
(I've been thinking recently of how to get us on more of a standard SQL grammar, I want black for SQL. 😄 )

Reply:
Making this uppercase is a good change 👍

Reply:
It doesn't matter too much here, but if you're debugging and seeing the full resulting SQL it helps a bit, I think.

Reply:
I had come across https://github.com/pgjones/flake8-sql, but it appears abandoned (and doesn't support Python >= 3.9 it seems).

Reply:
c392efc27